src/timeline-window.ts

```diff
@@ -135,13 +135,14 @@ export class TimelineWindow {
         // which is important to keep room-switching feeling snappy.
         if (this.timelineSet.getTimelineForEvent(initialEventId)) {
             initFields(this.timelineSet.getTimelineForEvent(initialEventId));
-            return Promise.resolve();
+            return;
         } else if (initialEventId) {
-            return this.client.getEventTimeline(this.timelineSet, initialEventId)
+            await this.client.getEventTimeline(this.timelineSet, initialEventId)
                 .then(initFields);
```

Comment:
`this.client.getEventTimeline(...).then(initFields)` is already a `Promise<void>`?

Reply:
Sorry, I think I confused my sentiment with [this spot](https://github.com/matrix-org/matrix-react-sdk/blob/f5f79158ed5a950626bc89439b8319e362a75de1/src/components/structures/LoggedInView.tsx#L356) (which is totally different, oops) that I mentioned in https://github.com/matrix-org/matrix-js-sdk/pull/2521#discussion_r920214132. I'm trying to get back to speed with everything going on in that stale PR.

And in order to refactor to a fire and forget `getEventTimeline` pattern here, I'd need to adjust how we `initFields` is done here.

I'll have a think on this one whether it's relevant. Good shout!